core/app/models/spree/stock/coordinator.rb

```diff
@@ -66,12 +65,19 @@ def build_packages(packages = Array.new)
 
       private
 
-      def unallocated_inventory_units
-        inventory_units - @preallocated_inventory_units
+      def stock_location_variant_ids
+        location_variant_ids = StockItem.where(variant_id: unallocated_variant_ids).joins(:stock_location).merge(StockLocation.active).pluck(:stock_location_id, :variant_id)
```

Comment:
@BenMorganIO I don't need stock_location_ids and variant_ids separately, I need them associated with each other.  And I need them in a specific enough way (for performance) that it doesn't seem good to move this into the model. Everything here is already using a scope or else is the way it is for perf reasons.